visualization/app/codeCharta/util/algorithm/streetLayout/streetViewHelper.ts

```diff
@@ -0,0 +1,93 @@
+import { Vector3 } from "three"
+import { CodeMapNode, Node, State } from "../../../codeCharta.model"
+import { getMarkingColor, isLeaf } from "../../codeMapHelper"
+import { getBuildingColor, getIncomingEdgePoint, isNodeFlat, isVisible, TreeMapHelper } from "../treeMapLayout/treeMapHelper"
+
+function calculateSize(node: CodeMapNode, metricName: string) { // TODO if it is same as countNodes in treeMapHelper.ts
+	let totalSize = node.attributes[metricName] || 0
+
+	if (totalSize === 0 && node.children && node.children.length > 0) {
+		for (const child of node.children) {
+			totalSize += calculateSize(child, metricName)
+		}
+	}
+	return totalSize
+}
+
+function mergeDirectories(node: CodeMapNode, metricName: string): CodeMapNode {
+	let mergedNode = node
+	for (const child of node.children) {
+		if (!isLeaf(child)) {
+			const nodeSize = calculateSize(node, metricName)
+			const childSize = calculateSize(child, metricName)
+			if (nodeSize === childSize) {
+				const nodeName = mergedNode.name
+				mergedNode = child
+				mergedNode.name = `${nodeName}/${child.name}`
+				break
+			}
```

Comment:
```suggestion
			}
			break
```

This is in combination with the comment above. If any child is not of size zero but also not the total nodeSize, it's impossible to have any other child that would fit the requirement. At least if I understood the logic correct.

Reply:
can't we get a child with a 0 metric value ? If not then I agree

Reply:
That's dealt with in my comment above :) https://github.com/MaibornWolff/codecharta/pull/1585#discussion_r542410449